main.go

```diff
@@ -42,45 +68,77 @@ func main() {
 	})
 
 	var data PromData
+	var ep Endpoint
 	data.Logger = logger
 	for _, a := range addresses {
+		logger.Debug("Parsing: ", a)
 		url, err := urlx.Parse(a)
 		if err != nil {
 			// we could not do parse address url
-			// TODO: logging
 			logger.Warn("could not parse address due to", err)
 			continue
 		}
-		data.URLs = append(data.URLs, url)
+		ep.Set(url)
+		data.EndPoints = append(data.EndPoints, ep)
 	}
 
+	go data.EndpointsProbe()
+
 	http.HandleFunc("/", data.PrometheusProxy) // set router
 	err := http.ListenAndServe(":9090", nil)   // set listen port
 	if err != nil {
 		log.Fatal("ListenAndServe: ", err)
 	}
 }
 
+func (p *PromData) EndpointsProbe() {
+	// we don't want to probe urls
+	logger := p.Logger
+	t, _ := time.ParseDuration("5m")
+	ticker := time.NewTicker(t)
+	m, _ := url.Parse("/metrics")
+	for _ = range ticker.C {
+		// every t period, do
+		for i, ep := range p.EndPoints {
+			url := ep.URL.ResolveReference(m).String()
+			_, err := http.Get(url)
+			if err != nil {
+				logger.Warning(ep.URL.String(), " is DOWN")
+				p.EndPoints[i].Disable()
+				continue
```

Comment:
Instead of `continue`, consider a switch statement?

Reply:
could you please provide an example? I don't see a way of using switch here :(

Reply:
@ahammond could you please provide an example? I don't see a way of using switch here :(